extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -144,13 +203,15 @@ function BaseCarouselWithRef(
       )}
       <Scroller
         advanceCount={advanceCount}
+        autoAdvanceCount={autoAdvanceCount}
         loop={loop}
         mixedLength={mixedLength}
         restingIndex={currentSlide}
         setRestingIndex={setRestingIndex}
         snap={snap}
         ref={scrollRef}
-        onTouchStart={() => setHadTouch(true)}
+        onTouchStart={() => setInteraction(Interaction.TOUCH)}
+        onMouseEnter={() => setInteraction(Interaction.MOUSE)}
```

Comment:
`onMouseEnter` seems like a very "greedy" interception. I might not actually do anything with the mouse, just hovering somewhere. Do we really want to change interaction model in this case?

Reply:
This was the recommendation from #29557, which encourages checking `mouseenter`: 
> We further recommend pausing the carousel as soon as the user's mouse/pointer is over the carousel, and when any of the controls or content inside the carousel receive keyboard focus. This will make sure that content doesn't change "under the user's feet" as they navigate the page.

Though that brings up I should also check for focus, I'll add another handler for this.

Another interesting point from that thread:
> One additional suggestion would be to take advantage of the prefers-reduced-motion media query. If the user has indicated a preference for reduced motion/animations, the autoplaying carousel could be paused by default (with the option for users to explicitly enable autoplay with the "Play/Pause" control if they wish).

Can we set a smart default to `autoAdvance = !window.matchMedia('(prefers-reduced-motion: reduce)') && autoAdvance`?